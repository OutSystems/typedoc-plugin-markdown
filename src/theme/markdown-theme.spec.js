const { Application } = require('typedoc');
const { MarkdownTheme } = require('../../dist/theme/markdown-theme');

describe(`[theme] markdown theme`, () => {
  test('should compile', () => {
    const app = new Application({
      mode: 'Modules',
      module: 'CommonJS',
      target: 'ES5',
    });
    const theme = new MarkdownTheme(app.renderer, './dist/theme', {
      mode: 'Modules',
      module: 'CommonJS',
      target: 'ES5',
    });
    const project = app.convert(app.expandInputFiles(['./test/src']));
    const urlMappings = theme.getUrls(project);
    const urls = urlMappings.map(urlMapping => {
      return urlMapping.url;
    });
    expect(urls).toMatchSnapshot();
  });
});
