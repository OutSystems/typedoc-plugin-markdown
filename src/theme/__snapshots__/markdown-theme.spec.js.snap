// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[theme] markdown theme should compile 1`] = `
Array [
  "README.md",
  "modules/_access_.md",
  "modules/_access_.privatemodule.md",
  "classes/_access_.privateclass.md",
  "modules/_classes_.md",
  "classes/_classes_.baseclass.md",
  "classes/_classes_.genericclass.md",
  "classes/_classes_.internalclass.md",
  "classes/_classes_.nongenericclass.md",
  "classes/_classes_.subclassa.md",
  "classes/_classes_.subclassb.md",
  "interfaces/_classes_.inameinterface.md",
  "interfaces/_classes_.iprintinterface.md",
  "interfaces/_classes_.iprintnameinterface.md",
  "modules/_default_export_.md",
  "classes/_default_export_.defaultexportedclass.md",
  "classes/_default_export_.notexportedclassname.md",
  "modules/_destructuring_.md",
  "modules/_doc_comments_.md",
  "modules/_enumerations_.md",
  "enums/_enumerations_.directions.md",
  "enums/_enumerations_.size.md",
  "modules/_flattened_.md",
  "classes/_flattened_.flattenedclass.md",
  "interfaces/_flattened_.bar.md",
  "interfaces/_flattened_.foo.md",
  "modules/_functions_.md",
  "modules/_functions_.modulefunction.md",
  "modules/_generics_.md",
  "classes/_generics_.myfirstclass.md",
  "classes/_generics_.myfourthclass.md",
  "classes/_generics_.mysecondclass.md",
  "classes/_generics_.mythirdclass.md",
  "interfaces/_generics_.a.md",
  "interfaces/_generics_.ab.md",
  "interfaces/_generics_.abnumber.md",
  "interfaces/_generics_.abstring.md",
  "interfaces/_generics_.b.md",
  "modules/_interfaces_.md",
  "modules/_interfaces_.interfaces.md",
  "interfaces/_interfaces_.interfaces.clockconstructor.md",
  "interfaces/_interfaces_.interfaces.clockinterface.md",
  "interfaces/_interfaces_.interfaces.shape.md",
  "interfaces/_interfaces_.interfaces.square.md",
  "interfaces/_interfaces_.interfaces.squareconfig.md",
  "interfaces/_interfaces_.interfaces.stringarray.md",
  "interfaces/_interfaces_.interfaces.surface.md",
  "classes/_interfaces_.color.md",
  "classes/_interfaces_.vector.md",
  "modules/_variables_.md",
]
`;
